cairo_vm_hints/src/hints/lib/verifiers/evm/account_verifier.rs

```diff
@@ -0,0 +1,230 @@
+    vm.write_arg(proof_ptr, &proof)?;
+
+    Ok(())
+}
+
+pub const HINT_ACCOUNT_PROOF_LEN: &str = "len(proof.proof)";
+
+pub fn hint_account_proof_len(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let proof = batch.accounts[idx].proofs[idx].clone();
+
+    insert_value_into_ap(vm, Felt252::from(proof.proof.len()))
+}
+
+pub const HINT_ACCOUNT_PROOF_BLOCK_NUMBER: &str = "proof.block_number";
+
+pub fn hint_account_proof_block_number(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let proof = batch.accounts[idx].proofs[idx].clone();
+
+    insert_value_into_ap(vm, Felt252::from(proof.block_number))
+}
+
+pub const HINT_ACCOUNT_PROOF_BYTES_LEN: &str = "segments.write_arg(ids.proof_bytes_len, proof.proof_bytes_len)";
+
+pub fn hint_account_proof_bytes_len(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let proof = batch.accounts[idx].proofs[idx].clone();
+
+    insert_value_from_var_name(
+        vars::ids::PROOF_BYTES_LEN,
+        MaybeRelocatable::Int(Felt252::from(proof.proof_bytes_len)),
+        vm,
+        &hint_data.ids_data,
+        &hint_data.ap_tracking,
+    )
+}
+
+pub const HINT_GET_MPT_PROOF: &str = "segments.write_arg(ids.mpt_proof, [int(x, 16) for x in proof.proof])";
+
+pub fn hint_get_mpt_proof(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let proof = batch.accounts[idx].proofs[idx].clone();
+    let mpt_proof_ptr = get_ptr_from_var_name(vars::ids::MPT_PROOF, vm, &hint_data.ids_data, &hint_data.ap_tracking)?;
+
+    let proof_bytes: Vec<u8> = proof.proof.iter().map(|x| x.to_bytes_be()).flatten().collect();
```

Comment:
(similar) why we convert proof felts to bytes then back to felts to insert_value?